portal-api/src/v1/users/sign-in-link.controller.js

```diff
@@ -1,10 +1,45 @@
+const { LOGIN_STATUSES } = require('../../constants');
+const utils = require('../../crypto/utils');
+const { updateLastLogin } = require('./controller');
+const { sanitizeUser } = require('./sanitizeUserData');
+
 class SignInLinkController {
   #signInLinkService;
 
   constructor(signInLinkService) {
     this.#signInLinkService = signInLinkService;
   }
 
+  async validateSignInLink(req, res) {
```

Comment:
I'd argue we do more than validate a sign in link, we complete the login and issue a full JWT. I see why you've gone for this name because validating the sign in link is the main step but if this was a 1FA login endpoint we wouldn't call it `validateUsernameAndPassword` which makes me think we might be slightly better off with a different name

Reply:
I know you've not changed the URL in this PR but maybe the same idea applies to the URL, which is currently
```
POST /users/me/validate-sign-in-link/:signInToken
```

Given the endpoint that creates the tokens is `POST /users/me/sign-in-link`, maybe we want this endpoint to be something like

```
POST /users/me/sign-in-link/:signInToken/login
```

What do you think?

Reply:
let me know what you think about the new name

Reply:
I like it, thanks @AlexBramhill